### PR TITLE
BAH-2453 | Address & contact information are not synced in crater.

### PR DESCRIPTION
**Pull request by atish160384**
- Configured and synced address and phone number when creating patient in Bahmni so that it will be visible in Crater.

### CODE REVIEW

File: feed-integration-webapp/src/main/java/org/bahmni/module/feedintegration/crater/CraterAPIClient.java [hunk 1]
```diff
@@ -20,165 +34,196 @@
...
+			httpClient.close();
+			response.close();
+			return ((result.charAt(0) == '1') ? true : false);
+		}
+	}
+
+	public void create_customer(OpenMRSPatientFullRepresentation patientFR, String auth) throws Exception {
+		String name = patientFR.getPerson().getPreferredName().getPreferredFullName();
+		String uuid = getuuid(patientFR);
+		List<NameValuePair> nameValuePairs = new ArrayList<NameValuePair>();
```
Comment (gsluthra) [inline feed-integration-webapp/src/main/java/org/bahmni/module/feedintegration/crater/CraterAPIClient.java]: The code for constructing nameValuePairs in create_cust() and update_cust() is the same. Plz extract it into a constructCustomerFromOpenmrsPatient() method and use that across. That way its also easier to write a unit test for that function.

File: feed-integration-webapp/src/main/java/org/bahmni/module/feedintegration/crater/CraterAPIClient.java [hunk 2]
```diff
@@ -20,165 +34,196 @@
...
+		((HttpRequestBase) request).setURI(uri);
+		logger.info("URI :{}",uri);
+		request.addHeader("Content-Type", content_type);
+		request.addHeader("Accept", accept);
+		request.addHeader("company", company);
+		request.addHeader("Authorization", "Bearer " + auth);
+		response = httpClient.execute(request);
+		logger.info("Status Code  : {}", response.getStatusLine().getStatusCode());
+		response.close();
+		httpClient.close();
```
Comment (gsluthra) [inline feed-integration-webapp/src/main/java/org/bahmni/module/feedintegration/crater/CraterAPIClient.java]: This code is also duplicate and can be extracted into sendRequestToCrate( ) method, and reused across both create() and update() methods, by passing the nameValuePairs as params.

**Review by gsluthra — CHANGES_REQUESTED** [review]

File: feed-integration-webapp/src/main/java/org/bahmni/module/feedintegration/crater/CraterAPIClient.java [hunk 3]
```diff
@@ -20,165 +34,170 @@
...
+		
+	}
+	
+	public void create_customer(OpenMRSPatientFullRepresentation patientFR, String auth) throws Exception {
+		String name = patientFR.getPerson().getPreferredName().getPreferredFullName();
+		String uuid = getuuid(patientFR);
+		OpenMRSPersonAddress address = patientFR.getPerson().getPreferredAddress();
+		List<OpenMRSPersonAttribute> attribute=patientFR.getPerson().getAttributes();
+		List<NameValuePair>nameValuePairs=constructCustomerFromOpenmrsPatient(name,uuid,currencyId,address,attribute);
+		CloseableHttpResponse response = null;
```
Comment (gsluthra) [inline feed-integration-webapp/src/main/java/org/bahmni/module/feedintegration/crater/CraterAPIClient.java]: Why not also create the RESPONSE object in the sendRequestToCrater( ) method?

File: feed-integration-webapp/src/main/java/org/bahmni/module/feedintegration/crater/CraterAPIClient.java [hunk 4]
```diff
@@ -20,165 +34,170 @@
...
+	public void update_customer(OpenMRSPatientFullRepresentation patientFR, String auth) throws Exception {
+		String name = patientFR.getPerson().getPreferredName().getPreferredFullName();
+		String uuid = getuuid(patientFR);
+		OpenMRSPersonAddress address = patientFR.getPerson().getPreferredAddress();
+		List<OpenMRSPersonAttribute> attribute=patientFR.getPerson().getAttributes();
+		List<NameValuePair>nameValuePairs=constructCustomerFromOpenmrsPatient(name,uuid,currencyId,address,attribute);
+		CloseableHttpResponse response = null;
+		String id = this.get_list_customers(uuid, auth);
+		HttpPut request = new HttpPut(crater_url + "/api/v1/customers/"+id);		
+		sendRequestToCreate(response,nameValuePairs,request);
```
Comment (gsluthra) [inline feed-integration-webapp/src/main/java/org/bahmni/module/feedintegration/crater/CraterAPIClient.java]: Why not also create the RESPONSE object in the sendRequestToCrater( ) method?

File: feed-integration-webapp/src/main/java/org/bahmni/module/feedintegration/crater/CraterAPIClient.java [hunk 5]
```diff
@@ -20,165 +34,170 @@
...
+		try (CloseableHttpResponse response = httpClient.execute(request)) {
+			HttpEntity entity = response.getEntity();
+			String result = EntityUtils.toString(entity);
+			httpClient.close();
+			response.close();
+			return ((result.charAt(0) == '1') ? true : false);
+		}
+	}
+	
+	public  void sendRequestToCreate(CloseableHttpResponse response,List<NameValuePair> nameValuePairs,HttpRequestBase request) throws Exception {
```
Comment (gsluthra) [inline feed-integration-webapp/src/main/java/org/bahmni/module/feedintegration/crater/CraterAPIClient.java]: Should be called, sendRequestToCrater( )

**Review by gsluthra — CHANGES_REQUESTED** [review]

File: feed-integration-webapp/src/main/java/org/bahmni/module/feedintegration/crater/CraterAPIClient.java (hunk 4 above)
Comment (atish160384) [reply feed-integration-webapp/src/main/java/org/bahmni/module/feedintegration/crater/CraterAPIClient.java]: done

File: feed-integration-webapp/src/main/java/org/bahmni/module/feedintegration/crater/CraterAPIClient.java (hunk 3 above)
Comment (atish160384) [reply feed-integration-webapp/src/main/java/org/bahmni/module/feedintegration/crater/CraterAPIClient.java]: done

**Review by gsluthra — APPROVED** [review]

**Comment by binduak** [comment]
This PR is no longer required. Closing the PR !

**Pull request closed without merging**